feat(eslint-plugin): added related-getter-setter-pairs rule (#10192)
typescript-eslint/typescript-eslint: packages/eslint-plugin/docs/rules/related-getter-setter-pairs.mdx

---
description: 'Enforce that `get()` types should be assignable to their equivalent `set()` type.'
---

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

> 🛑 This file is source code, not the primary documentation location! 🛑
>
> See **https://typescript-eslint.io/rules/related-getter-setter-pairs** for documentation.

TypeScript allows defining different types for a `get` parameter and its corresponding `set` return.

> Prior to TypeScript 4.3, the types didn't have to be related.

Reviewer: ```suggestion
Prior to TypeScript 4.3, the types had to be identical.
```

Right?
